testsuite/MDAnalysisTests/lib/test_util.py

```diff
@@ -444,128 +437,125 @@ def _fill_cache(self, name, value):
         self._cache[name] = value
 
 
-class TestCachedDecorator(TestCase):
-    def setUp(self):
-        self.obj = Class_with_Caches()
+class TestCachedDecorator(object):
+    @pytest.fixture()
+    def obj(self):
+        return Class_with_Caches()
 
-    def tearDown(self):
-        del self.obj
-
-    def test_val1_lookup(self):
-        self.obj._clear_caches()
-        assert_equal('val1' in self.obj._cache, False)
-        assert_equal(self.obj.val1(), self.obj.ref1)
-        ret = self.obj.val1()
-        assert_equal('val1' in self.obj._cache, True)
-        assert_equal(self.obj._cache['val1'], ret)
-        assert_equal(self.obj.val1() is self.obj._cache['val1'], True)
-
-    def test_val1_inject(self):
+    def test_val1_lookup(self, obj):
+        obj._clear_caches()
+        assert ('val1' in obj._cache) == False
+        assert obj.val1() == obj.ref1
+        ret = obj.val1()
+        assert ('val1' in obj._cache) == True
+        assert obj._cache['val1'] == ret
+        assert obj.val1() is obj._cache['val1'] == True
+
+    def test_val1_inject(self, obj):
         # Put something else into the cache and check it gets returned
         # this tests that the cache is blindly being used
-        self.obj._clear_caches()
-        ret = self.obj.val1()
-        assert_equal('val1' in self.obj._cache, True)
-        assert_equal(ret, self.obj.ref1)
+        obj._clear_caches()
+        ret = obj.val1()
+        assert ('val1' in obj._cache) == True
```

Comment:
as @jbarnoud mentioned you don't need to do `assert foo == True` just using `assert foo` is enough